packages/engine/src/world/SketchManager.ts

```diff
@@ -16,13 +16,23 @@ type SketchInstance = {
   root?: Group
 
   getPasses?: (engineScene: EngineScene) => Pass[]
+
+  /**
+   * Called when the sketch is removed from the scene.
+   * It should clean up any resources, event listeners, or references it holds.
+   */
+  dispose(): () => void
 }
 
 export class SketchManager {
   private sketchInstances: { [id: string]: SketchInstance } = {}
 
-  private createSketch = (instanceId: string, module: SketchModule): SketchInstance => {
-    const sketch = new module(getDebugScene())
+  private createSketch = (
+    instanceId: string,
+    module: SketchModule,
+    scene: EngineScene,
+  ): SketchInstance => {
+    const sketch = new module(scene)
```

Comment:
passing the EngineScene through instead, since we get it in the function prior anyways

Reply:
Makes sense! We're gonna need a pattern like this later anyway when we have multiple scenes